### PR TITLE
Task self reflection add assertion

### LINKED ISSUE #75: [Task] Add assertions to self reflection feature
**Task**
What is the purpose this feature aims to achieve?  
Adding assertions to check for assumptions. 
Refer to #62 

**Is your task request related to a problem? Please describe.**
Failing to fulfill the assumptions might crash the program. 

**Describe the solution you'd like**
Add assertions at key junctions to check verify the assumptions.

### PR DESCRIPTION
**Pull request by wenxin-c**
This fixes #75 
Adding assertions to self reflection feature to make a more defensive.

### CODE REVIEW

**Review by nichyjt — COMMENTED** [review]
I have some comments regarding your use of exceptions and assertions and a few readability nits. For your consideration!

File: src/main/java/wellnus/reflection/SelfReflection.java [hunk 1]
```diff
@@ -8,18 +10,18 @@
...
+                + "| (___ | |__  | |    | |__    | |__) | |__  | |__  | "
+                + "|    | |__ | |       | |    | || |  | |  \\| |\n"
+                + " \\___ \\|  __| | |    |  __|   |  _  /|  __| |  __| "
+                + "| |    |  __|| |       | |    | || |  | | . ` |\n"
+                + " ____) | |____| |____| |      | | \\ \\| |____| |    "
+                + "| |____| |___| |____   | |   _| || |__| | |\\  |\n"
+                + "|_____/|______|______|_|      |_|  \\_\\______|_|    "
+                + "|______|______\\_____|  |_|  "
+                + "|_____\\____/|_| \\_|\n";
     private static final String GREETING_MESSAGE = "Welcome to WellNUS++ Self Reflection section:D"
```
[the remark this answers is not in the document]
Comment (nichyjt) [reply src/main/java/wellnus/reflection/SelfReflection.java:25]: This is for professionalism! Emojis are fine but the lack of spacing might make the output appear too informal/not taking this seriously.

File: src/main/java/wellnus/reflection/SelfReflection.java [hunk 2]
```diff
@@ -39,18 +41,21 @@ public class SelfReflection {
         "What matters to me most right now?"
     };
 
+    private static final int TOTAL_NUM_QUESTIONS = 10;
+    private static final String TOTAL_NUM_QUESTION_ASSERTIONS = "The total number of questions is 10.";
```
[the remark this answers is not in the document]
Comment (wenxin-c) [reply src/main/java/wellnus/reflection/SelfReflection.java:45]: i'll leave a coomment there to update/change at later stages

File: src/main/java/wellnus/reflection/ReflectionQuestion.java [hunk 3]
```diff
@@ -1,12 +1,20 @@
...
+    private static final int EMPTY_QUESTION = 0;
+    private static final String EMPTY_QUESTION_MSG = "Question description cannot be empty.";
     private String questionDescription;
 
-    public ReflectionQuestion(String questionDescription) {
+    public ReflectionQuestion(String questionDescription) throws EmptyReflectionQuestionException {
+        if (questionDescription.length() == EMPTY_QUESTION) {
+            throw new EmptyReflectionQuestionException(EMPTY_QUESTION_MSG);
+        }
+        assert questionDescription.length() > EMPTY_QUESTION : EMPTY_QUESTION_MSG;
```
[the remark this answers is not in the document]
Comment (wenxin-c) [reply src/main/java/wellnus/reflection/ReflectionQuestion.java:17]: hmmm i'll leave the assertion here and remove the exception

**Review by nichyjt — APPROVED** [review]
LGTM! These changes help make the codebase tighter and more cohesive.

**Pull request merged**